### PR TITLE
Issue #21596: Post Documents in a single step

### PR DESCRIPTION
**Pull request by andersdd**
Add ability to post from within Misc Invoice and Misc Voucher documents.  Significant time saving for users who wish to utilise this feature.

### CODE REVIEW

File: guiclient/invoice.ui [hunk 1]
```diff
@@ -44,6 +44,13 @@ to be bound by its terms.</comment>
      </item>
      <item>
       <layout class="QHBoxLayout">
+     <item>
+      <widget class="QCheckBox" name="_postInvoice">
+       <property name="text">
```
Comment (gilmoskowitz) [inline guiclient/invoice.ui]: This should be an XCheckBox

File: guiclient/invoice.cpp [hunk 2]
```diff
@@ -728,6 +736,104 @@ bool invoice::save()
...
+                __FILE__, __LINE__);
+     }
+     else if (!xrate.first() || xrate.value("curr_rate").isNull())
+     {
+       systemError(this, tr("Could not post Invoice #%1 because of a missing exchange rate.")
+				.arg(_invoiceNumber->text()));
+     }
+  }
+
+  unpostedPost.exec("BEGIN;");	// because of possible lot, serial, or location distribution cancelations
```
Comment (gilmoskowitz) [inline guiclient/invoice.cpp]: The transaction appears to start too late - the current structure allows posting the invoice without distributing inventory. With the current separation, how & when will the inventory get distributed if there's an error or the user cancels the distribution?

File: guiclient/invoice.ui [hunk 3]
```diff
@@ -44,6 +44,13 @@ to be bound by its terms.</comment>
      </item>
      <item>
       <layout class="QHBoxLayout">
+     <item>
+      <widget class="QCheckBox" name="_postInvoice">
```
Comment (gilmoskowitz) [inline guiclient/invoice.ui]: This should be an `XCheckBox`.

File: guiclient/invoice.ui [hunk 4]
```diff
@@ -1472,8 +1481,6 @@ to be bound by its terms.</comment>
   <tabstop>_salesrep</tabstop>
   <tabstop>_commission</tabstop>
   <tabstop>_taxzone</tabstop>
-  <tabstop>_cust</tabstop>
-  <tabstop>_copyToShipto</tabstop>
```
Comment (gilmoskowitz) [inline guiclient/invoice.ui]: Was this change in tab order made consciously?

File: guiclient/miscVoucher.ui [hunk 5]
```diff
@@ -40,6 +40,13 @@ to be bound by its terms.</comment>
       </spacer>
      </item>
      <item>
+      <widget class="QCheckBox" name="_postVoucher">
```
Comment (gilmoskowitz) [inline guiclient/miscVoucher.ui]: This should be an `XCheckBox`.

File: guiclient/invoice.ui (hunk 4 above)
Comment (andersdd) [inline guiclient/invoice.ui]: Yes it was - the first thing required on an invoice is the customer, but that was not the starting widget

File: guiclient/invoice.cpp (hunk 2 above)
Comment (andersdd) [inline guiclient/invoice.cpp]: Interesting given I copied this logic from the post invoice function in the Unposted Invoices screen.  I will check further. May have to apply the logic elsewhere.

Comment (gilmoskowitz) [inline guiclient/invoice.cpp]: hmmm... Then I might be wrong about the structure of the transaction. @garyhgohoos - do you have an opinion here?

Comment (garyhgohoos) [inline guiclient/invoice.cpp]: the sum and xrate queries are performed outside of the transaction.  the post and distribution are inside of the begin/commit of the transaction.  looks ok to me.

Comment (garyhgohoos) [inline guiclient/invoice.cpp]: post.prepare is outside, but post.exec is inside

Comment (gilmoskowitz) [inline guiclient/invoice.cpp]: Ahhh... Thanks @garyhgohoos !

**Pull request merged**